bok_choy/a11y/axs_ruleset.py

```diff
@@ -0,0 +1,239 @@
+"""
+Interface for using the google accessibility ruleset.
+See: https://github.com/GoogleChrome/accessibility-developer-tools
+"""
+import json
+import logging
+import os
+import requests
+
+from collections import namedtuple
+from textwrap import dedent
+
+from .a11y_audit import A11yAudit, A11yAuditConfig, AccessibilityError
+
+log = logging.getLogger(__name__)
+AuditResults = namedtuple('AuditResults', 'errors, warnings')
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+
+
+class AxsAuditConfig(A11yAuditConfig):
+    """
+    The `AxsAuditConfig` object defines the options available when
+    running an `AxsAudit`.
+    """
+    def __init__(self, *args, **kwargs):
+        super(AxsAuditConfig, self).__init__(*args, **kwargs)
+        self.rules_file = None
+        self.scope, self.rules_to_run, self.rules_to_ignore = None, None, None
+        self.rules_file = os.path.join(
+            os.path.split(CUR_DIR)[0],
+            'vendor/google/axs_testing.js'
+        )
+        self.set_rules([], ignore=False)
+        self.set_rules([], ignore=True)
+        self.set_scope()
+
+    def set_rules(self, rules, ignore=False, include_type="rule"):
+        """
+        Sets `rules_to_run` or `rules_to_ignore` to the passed list of rule IDs.
+
+        Args:
+
+            rules: List of rules to ignore or run for accessibility errors on
+                the page. If `ignore=False`, an empty list means to check for
+                all available rules and `None` means that no audit should be
+                done for this page. If `ignore=True`, an empty list means to
+                run rules all otherwise enabled rules and any rules listed here
+                will be ignored even if they are also specified as rules to run.
+                See https://github.com/GoogleChrome/accessibility-developer-tools/tree/master/src/audits
+            ignore (bool) (optional): If you want to ignore the specified rules
+                set this to `True`.  By default this is `False`.
+
+        Examples:
+
+            To check only `badAriaAttributeValue`::
+
+                page.a11y_audit.config.set_rules(
+                    ['badAriaAttributeValue']
+                )
+
+            To check all rules except `badAriaAttributeValue`::
+
+                page.a11y_audit.config.set_rules(
+                    ['badAriaAttributeValue'],
+                    ignore=True
+                )
+        """
+        if ignore:
+            self.rules_to_ignore = rules
+        else:
+            self.rules_to_run = rules
+
+    def set_scope(self, include=None, exclude=None):
+        """
+        Sets `scope`, the "start point" for the audit.
+
+        Args:
+
+            include: A selector specifying the element which contains the
```

Comment:
We need to normalize this method across the 2 implementations. That is, AxsAuditConfig requires the set_scope method's include argument to be a string representing a css selector, and AxeCoreAuditConfig requires a list of strings. So if for example you have a testcase class that initializes the audit type in the setup method which has a bunch of testcases that set the include scope, it will not be straightforward to switch back and forth between the rulesets. E.g. you can't ddt to run them both.